Frameworks/Foundation/NSData.mm

```diff
@@ -396,42 +388,29 @@ - (instancetype)initWithContentsOfFile:(NSString*)filename options:(NSDataReadin
         return nil;
     }
 
-    char* fname = (char*)[filename UTF8String];
-
-    TraceVerbose(TAG, L"NSData extended-opening %hs", fname);
-    EbrFile* fpIn = EbrFopen(fname, "rb");
-    if (fpIn) {
-        auto closeFile = wil::ScopeExit([&]() { EbrFclose(fpIn); });
-
-        EbrFseek(fpIn, 0, SEEK_END);
-        size_t length = EbrFtell(fpIn);
-        EbrFseek(fpIn, 0, SEEK_SET);
-
-        if (length) {
-            // The created NSData takes ownership of freeing this
-            woc::unique_iw<uint8_t> copiedBytes(static_cast<uint8_t*>(IwMalloc(length)));
-            if (!copiedBytes.get()) {
-                if (error) {
-                    *error = [NSError errorWithDomain:@"NSData" code:100 userInfo:nil];
-                }
-                [self release];
-                return nil;
-            }
-
-            size_t fileLength = EbrFread(copiedBytes.get(), 1, length, fpIn);
-            self = [self initWithBytesNoCopy:copiedBytes.release() length:fileLength];
-        }
+    NSInputStream* inputStream = [NSInputStream inputStreamWithFileAtPath:filename];
+    [inputStream open];
 
-    } else {
-        TraceVerbose(TAG, L"NSData couldn't open %hs for read (extended)", fname);
+    if (NSStreamStatusOpen != inputStream.streamStatus) {
+        TraceVerbose(TAG, L"NSData couldn't open %hs for read (extended)", filename);
         if (error) {
             *error = [NSError errorWithDomain:@"NSData" code:100 userInfo:nil];
         }
         [self release];
         return nil;
     }
 
-    return self;
+    auto closeFile = wil::ScopeExit([&]() { [inputStream close]; });
+
+    uint8_t byteBuffer[4096] = {};
+    NSMutableData* tempData = [NSMutableData data];
+
+    while ([inputStream hasBytesAvailable]) {
+        auto result = [inputStream read:byteBuffer maxLength:_countof(byteBuffer)];
```

Comment:
>byteBuffer [](start = 40, length = 10)

you can also just get the mutableBytes of the NSMutableData and read into it directly.
If you need to grow the capacity, you can do it via increaseLength